src/GafferUI/Handle.cpp

```diff
@@ -71,6 +71,47 @@ using namespace IECore;
 using namespace IECoreScene;
 using namespace GafferUI;
 
+const float g_planarToLinearDragThreshold = 0.15f;  // Approx. 9 degrees from perpendicular
+
+namespace
+{
+
+struct WorldGadgetDragData
+{
+	V3f worldOrigin;
+	V3f worldAxis0;
+	V3f worldAxis1;
+	Line3f worldLine;
+	Plane3f worldPlane;
+};
+
+void worldGadgetDragData(
+	const Gadget *gadget,
+	const V3f &origin,
+	const V3f &axis0,
+	const V3f &axis1,
+	const DragDropEvent &dragEvent,
+	WorldGadgetDragData &result
+)
+{
+	const M44f transform = gadget->fullTransform();
+	result.worldOrigin = origin * transform;
+	transform.multDirMatrix( axis0, result.worldAxis0 );
+	transform.multDirMatrix( axis1, result.worldAxis1 );
+
+	result.worldLine = Line3f(
+		dragEvent.line.p0 * transform,
+		dragEvent.line.p1 * transform
+	);
+	result.worldPlane = Plane3f(
+		result.worldOrigin,
+		result.worldOrigin + result.worldAxis0,
+		result.worldOrigin + result.worldAxis1
```

Comment:
Gah, just lost a lengthy comment to webbrowser weirdness, I'll try to summarize over again:
I don't like using the 3 point version of the Plane3f constructor when we have directions to start with.
The Plane3f constructor just does:
```
normal = (point2 - point1) % (point3 - point1);
normal.normalize();
```
We can cancel out adding and subtracting worldOrigin and just do: `normal = ( worldAxis0 % worldAxis1 ).normalized()`, and in this case only the normal is used, so we don't need to store a Plane3f.

I would also say it would be better to compute the normal ourselves before passing it in to the temp Plane3f in updatePosition ... but this commit isn't touching that.

Reply:
Also a nice improvement, thanks for pointing that out. I removed the `Plane3f` use in the new code in https://github.com/GafferHQ/gaffer/pull/5281/commits/4cf983bd9790af0bd5f6f1b0b0af9c8c52580048.

In https://github.com/GafferHQ/gaffer/pull/5281/commits/3b696193e4074c399f803cda406288b35a10969f I strayed a bit out of scope of this PR and changed the `Plane3f` construction in `updatePosition()`. Not sure if it's a clear win as we drop a little calculation in exchange for another `V3f` stored in the class, but it's there for consideration.